com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/FunctionDescriptorNodeBuilder.cs

```diff
@@ -173,31 +173,15 @@ internal FunctionDescriptorNodeBuilder(FunctionDescriptor fd)
             ShaderContainer container,
             Registry registry)
         {
-            // Get the ShaderType for the first output port we find.
-            string outPortName = null;
-            foreach (var param in m_functionDescriptor.Parameters)
-            {
-                if (param.Usage == Usage.Out)
-                {
-                    outPortName = param.Name;
-                    break;
-                }
-            }
-            if (outPortName == null)
-            {
-                // No out port was found.
-                throw new Exception("No output port found for ");
-            }
-
-            data.TryGetPort(outPortName, out var port);
-            var shaderType = registry.GetShaderType((IFieldReader)port, container);
-
             // Get a builder from ShaderFoundry
             var shaderFunctionBuilder = new ShaderFunction.Builder(container, m_functionDescriptor.Name);
 
             // Set up the vars in the shader function.
             foreach (var param in m_functionDescriptor.Parameters)
             {
+                data.TryGetPort(param.Name, out var port);
+                var shaderType = registry.GetShaderType((IFieldReader)port, container);
```

Comment:
Because we were only getting the shaderType of the output port, all the `shaderFunctionBuilder.AddInput(shaderType, ...);` were using the same shaderType. This just makes it so we fetch the concrete type for each port.